File: Java/Implementations/Strategy/FlightMovement.java
import java.lang.Math;

public class FlightMovement implements Movement {
	private double speed;

	public FlightMovement(double speed) {
		this.speed = speed;
	}

	public String moveTo(Position from, Position to) {
		double distance = from.distanceTo(to);
		double time = distance / speed;
		return "flies from " + from + " to " + to
			+ ",\na distance of " + Utils.roundTo(distance, 2)
			+ ",\nat a speed of " + Utils.roundTo(this.speed, 2)
			+ ",\nwhich will take " + Utils.roundTo(time, 2) + " seconds";
	}
}

File: Haskell/Implementations/Strategy/FlightMovement.hs
module FlightMovement
    ( flightMovement
    ) where

import Movement (Movement, Position, distanceBetween)
import Util (roundTo)

flightMovement :: Double -> Movement
flightMovement speed from to = mconcat
    [ "flies from "
    , show from
    , " to "
    , show to
    , ",\na distance of "
    , round2 distance
    , ",\nat a speed of "
    , round2 speed
    , ",\nwhich will take "
    , round2 time
    , " seconds"
    ]
  where
    distance = distanceBetween from to
    time     = distance / speed
    round2   = show . roundTo 2
